pynamodb/models.py

```diff
@@ -542,16 +542,24 @@ def count(cls,
             non_key_attribute_classes=non_key_attribute_classes,
             filters=filters)
 
-        data = cls._get_connection().query(
-            hash_key,
-            index_name=index_name,
-            consistent_read=consistent_read,
-            key_conditions=key_conditions,
-            query_filters=query_filters,
-            limit=limit,
-            select=COUNT
-        )
-        return data.get(CAMEL_COUNT)
+        count_buffer = 0
+        last_evaluated_key = None
+        data = None
+        while not data or last_evaluated_key:
```

Comment:
The most minimal object that should be returned is `{'Count': 0}`, which should be truth-y here. This is a good point though, so I replaced this check with an explicit `started` bool to be more explicit.